### PR TITLE
Add an input gesture to temporarily disable all configuration profiles

### PR DESCRIPTION
**Pull request by LeonarddeR**
I realised that I never posted a pull request for this, or I missed that I did. In any case, 'here's an implementation for #4935

### CODE REVIEW

**Comment by feerrenrut** [comment]
What happens when a profile has already been triggered? Is the intention to revert to the default profile?

**Comment by leonardder** [comment]
[**@feerrenrut**](https://github.com/feerrenrut) commented on [30 aug. 2017 10:47 CEST](https://github.com/nvaccess/nvda/pull/6488#issuecomment-325925639 "2017-08-30T08:47:36Z - Replied by Github Reply Comments"):
> What happens when a profile has already been triggered? Is the intention to revert to the default profile?

Very good point there. It looks like there are special functions for enabling and disabling profile triggers, this pr only toggles the state in the configuration manager. This should be fixed to disable all triggered profiles as soon as you trigger.

**Review by feerrenrut — APPROVED** [review]

**Comment by leonardder** [comment]
I also noticed that the current code doesn't yet properly trigger profiles as soon as you reactivate them. I must have been asleep when I wrote this.

**Comment by feerrenrut** [comment]
oh oops, just saw your comment. In that case does this change need to be reverted?

**Comment by feerrenrut** [comment]
I have reverted this change from next. I will have to look at this again another day.

**Comment by leonardder** [comment]
I"m very sorry for the confusion. I added another commit and did some additional tests, and I belief this is now ready for real, that is, when you agree about the method used to trigger the profile(s) for the current application.

File: source/globalCommands.py [hunk 1]
```diff
@@ -2028,6 +2028,24 @@ def script_activateConfigProfilesDialog(self, gesture):
...
 
+	def script_toggleConfigProfileTriggers(self,gesture):
+		if config.conf.profileTriggersEnabled:
+			config.conf.disableProfileTriggers()
+			# Translators: The message announced when temporarily disabling all configuration profile triggers.
+			state = _("Configuration profile triggers disabled")
+		else:
+			config.conf.enableProfileTriggers()
+			# Explicitly trigger profiles for the current application.
+			mod=api.getForegroundObject().appModule
```
Comment (dkager) [inline source/globalCommands.py]: Missing space around the `=` sign.

File: source/globalCommands.py [hunk 2]
```diff
@@ -2028,6 +2028,24 @@ def script_activateConfigProfilesDialog(self, gesture):
...
+			trigger = mod._configProfileTrigger = appModuleHandler.AppProfileTrigger(mod.appName)
+			trigger.enter()
+			# Translators: The message announced when re-enabling all configuration profile triggers.
+			state = _("Configuration profile triggers enabled")
+		ui.message(state)
+	# Translators: Input help mode message for toggle configuration profile triggers command.
+	script_toggleConfigProfileTriggers.__doc__=_("Disables or enables all configuration profile triggers. Disabling takes effect until NVDA is restarted")
```
Comment (dkager) [inline source/globalCommands.py]: Maybe change this a bit:
"Toggles disabling of all configuration profile triggers. Disabling remains in effect until NVDA is restarted"

**Review by dkager — CHANGES_REQUESTED** [review]

File: source/globalCommands.py (hunk 1 above)
Comment (leonardder) [reply source/globalCommands.py]: Are there conventions for this? I'm happy for this to be changed, but I belief there are more cases in the code where equals isn't surrounded by spaces.

Comment (dkager) [reply source/globalCommands.py]: > Are there conventions for this?

No, but multiple ways of dealing with whitespace in a single function makes me sad. :)

**Review by dkager — APPROVED** [review]

**Comment by derekriemer** [comment]
If there are spaces around the surrounding = then put them, else no I'm pretty sure it's best practice to when we can, although that was a jamie Preference.

**Review by feerrenrut — APPROVED** [review]

**Pull request merged**